File: server/src/main/java/jk/kamoru/ground/flay/domain/FlayCondition.java
package jk.kamoru.ground.flay.domain;

import lombok.Data;

@Data
public class FlayCondition {

  public static enum Sort {
    STUDIO, OPUS, TITLE, ACTRESS, RELEASE, PLAY, RANK, LASTPLAY, LASTACCESS, LASTMODIFIED, SCORE, LENGTH, SHOT;
  }

  private Sort sort = Sort.RELEASE;
  private boolean reverse = false;

  private String studio;
  private String opus;
  private String title;
  private String actress;
  private String release;
  private int[] rank = new int[] { 0, 1, 2, 3, 4, 5 };
  private String search;
  private boolean withSubtitles = false;
  private boolean withFavorite = false;
  private boolean withNoFavorite = false;

}

File: server_node/src/domain/flay-condition.ts
/** FlayCondition 정렬 기준 (Java Sort enum 대응) */
export type FlaySort = 'STUDIO' | 'OPUS' | 'TITLE' | 'ACTRESS' | 'RELEASE' | 'PLAY' | 'RANK' | 'LASTPLAY' | 'LASTSHOT' | 'LASTACCESS' | 'LASTMODIFIED' | 'SCORE' | 'LENGTH' | 'SHOT';

/** Flay 검색/필터 조건 (FlayCondition 대응) */
export interface FlayCondition {
  sort: FlaySort;
  reverse: boolean;
  studio?: string;
  opus?: string;
  title?: string;
  actress?: string;
  release?: string;
  rank: number[];
  search?: string;
  withSubtitles: boolean;
  withFavorite: boolean;
  withNoFavorite: boolean;
}

/** 기본 FlayCondition */
export function createDefaultCondition(): FlayCondition {
  return {
    sort: 'RELEASE',
    reverse: false,
    rank: [0, 1, 2, 3, 4, 5],
    withSubtitles: false,
    withFavorite: false,
    withNoFavorite: false,
  };
}
